frontend/src/components/pages/Dashboard/EditDashboardSchedule.tsx

```diff
@@ -0,0 +1,137 @@
+import { ArrowBackIcon, EditIcon } from "@chakra-ui/icons";
+import {
+  Badge,
+  Button,
+  Container,
+  IconButton,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+
+import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import SchedulingAPIClient from "../../../APIClients/SchedulingAPIClient";
+import * as Routes from "../../../constants/Routes";
+import { DonorResponse } from "../../../types/DonorTypes";
+import { Schedule } from "../../../types/SchedulingTypes";
+
+const EditDashboard = (): JSX.Element => {
+  const { id } = useParams<{ id: string }>();
+  const history = useHistory();
+
+  const [currentSchedule, setCurrentSchedule] = useState<Schedule>();
+  const [currentDonor, setCurrentDonor] = useState<DonorResponse>();
+
+  const getScheduleData = async () => {
+    const response = await SchedulingAPIClient.getScheduleById(id);
+    setCurrentSchedule(response);
+  };
+
+  const getDonorData = async () => {
+    const response = await DonorAPIClient.getDonorById(
+      currentSchedule?.donorId ?? "",
+    );
+    setCurrentDonor(response);
+  };
+
+  useEffect(() => {
+    getScheduleData();
+    getDonorData();
```

Comment:
so with this current implementation, donor data is not shown because `getDonorData` does not wait for `getScheduleData` and currentSchedule is empty when `getDonorData` runs. One way to fix would be to do smth like the following:

```
const getScheduleDonorData = async () => {
    const response = await SchedulingAPIClient.getScheduleById(id);
    setCurrentSchedule(response);
    const donorResponse = await DonorAPIClient.getDonorById(
      response?.donorId ?? "",
    );
    setCurrentDonor(donorResponse);
  };

  useEffect(() => {
    getScheduleDonorData();
  }, [id]);
```